lib/board.rb

```diff
@@ -32,8 +37,45 @@ def free_positions
     flat_board.each_index.select { |index| empty?(flat_board[index]) }
   end
 
+  def content
+    board_string = ""
+
+    board.flatten.each.with_index do |value, index|
+      if !value.is_a?(Integer)
+        board_string << value
+      else
+        board_string << " "
+      end
+
+      index += 1
+      if  board_edge?(index) && index != @POSITION_MAX
+        board_string << ","
+      end
+
+    end
+
+    board_string
+  end
+
   private
 
+  def rows
+    board
+  end
+
+  def columns
+    board.transpose
+  end
+
+  def diagonals
```

Comment:
`[left_diagonal, right_diagonal]`?
What I'm trying to get at is instead of for-eaching over the arrays (`board.size.times`) and creating the resulting arrays, there are ways to `reduce` a new array out of an existing one.